cmd/skaffold/app/cmd/parse_config.go

```diff
@@ -92,6 +92,19 @@ func getConfigs(cfgOpts configOpts, opts config.SkaffoldOptions, r *record) ([]*
 	}
 	logrus.Debugf("parsed %d configs from configuration file %s", len(parsed), cfgOpts.file)
 
+	// validate that config names are unique if specified
+	seen := make(map[string]bool)
+	for _, cfg := range parsed {
+		cfgName := cfg.(*latest.SkaffoldConfig).Metadata.Name
+		if cfgName == "" {
+			continue
+		}
+		if seen[cfgName] {
+			return nil, fmt.Errorf("multiple skaffold configs named %q found in file %q", cfgName, cfgOpts.file)
```

Comment:
Maybe `seen` should be a map of `name` &rarr; file?  Then you could include at least two of the file nams.